processing/src/main/java/org/apache/druid/segment/serde/NullColumnPartSerde.java

```diff
@@ -0,0 +1,212 @@
+  public BitmapSerdeFactory getBitmapSerdeFactory()
+  {
+    return bitmapSerdeFactory;
+  }
+
+  @Nullable
+  @Override
+  public Serializer getSerializer()
+  {
+    return NOOP_SERIALIZER;
+  }
+
+  @Override
+  public Deserializer getDeserializer()
+  {
+    return (buffer, builder, columnConfig) -> {
+      builder
+          .setHasMultipleValues(false)
+          .setHasNulls(true)
+          .setFilterable(true)
+          .setBitmapIndex(Suppliers.ofInstance(bitmapIndex));
+      builder.setDictionaryEncodedColumnSupplier(Suppliers.ofInstance(nullDictionaryEncodedColumn));
+    };
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    NullColumnPartSerde partSerde = (NullColumnPartSerde) o;
+    return numRows == partSerde.numRows
+           && bitmapSerdeFactory.equals(partSerde.bitmapSerdeFactory);
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Objects.hash(numRows, bitmapSerdeFactory);
+  }
+
+  private final class NullDictionaryEncodedColumn implements DictionaryEncodedColumn<String>
+  {
+    // Get the singleton instance of DimensionSelector.NullDimensionSelectorHolder.NullDimensionSelector
+    // to reuse its dictionary lookup logic.
+    private final DimensionSelector nullDimensionSelector = DimensionSelector.constant(null);
+
+    @Override
+    public int length()
+    {
+      return numRows;
+    }
+
+    @Override
+    public boolean hasMultipleValues()
+    {
+      return false;
+    }
+
+    @Override
+    public int getSingleValueRow(int rowNum)
+    {
+      throw new RuntimeException("This method should not be called for null-only columns");
+    }
+
+    @Override
+    public IndexedInts getMultiValueRow(int rowNum)
+    {
+      throw new RuntimeException("This method should not be called for null-only columns");
+    }
+
+    @Nullable
+    @Override
+    public String lookupName(int id)
+    {
+      return nullDimensionSelector.lookupName(id);
+    }
+
+    @Override
+    public int lookupId(String name)
+    {
+      return nullDimensionSelector.idLookup().lookupId(name);
+    }
+
+    @Override
+    public int getCardinality()
+    {
+      return 1;
+    }
+
+    @Override
+    public DimensionSelector makeDimensionSelector(
+        ReadableOffset offset,
+        @Nullable ExtractionFn extractionFn
+    )
+    {
+      return DimensionSelector.constant(null, extractionFn);
+    }
+
+    @Override
+    public SingleValueDimensionVectorSelector makeSingleValueDimensionVectorSelector(
+        ReadableVectorOffset vectorOffset
+    )
+    {
+      return NilVectorSelector.create(vectorOffset);
+    }
+
+    @Override
+    public MultiValueDimensionVectorSelector makeMultiValueDimensionVectorSelector(
+        ReadableVectorOffset vectorOffset
+    )
+    {
+      throw new RuntimeException("This method should not be called for null-only columns");
+    }
+
```

Comment:
oops, I think you should implement `makeVectorValueSelector` and `makeVectorObjectSelector` here as well to return `NilVectorSelector` for missing numeric and complex columns

Reply:
I don't think they are strictly necessary in this PR since numeric columns don't use `NullColumnPartSerde` yet even when it's completely empty as it's noted in the "Future work" section of the PR description. If you agree, I'd like to add them in a follow-up PR when I fix numeric columns.